test.js

```diff
@@ -1,36 +1,136 @@
+	const openSpy = tq.spy();
+	const logger = tq.spy();
+
+	await tq.replace({
+		modulePath: new URL('cli.js', import.meta.url),
+		importMeta: import.meta,
+		localMocks: {
+			open: openSpy,
+			import: {console: {error: logger, log: logger}},
+			...packageJson && {
+				'package-json': tq.resolves(packageJson),
+			},
+		},
+		globalMocks: {
+			process: {
+				argv: [,, ...arguments_], // eslint-disable-line no-sparse-arrays,
+				env: {NO_COLOR: '1'},
+			},
+		},
+	});
+
+	const {flatCalls: urls} = tq.explain(openSpy);
+	t.deepEqual(urls.sort(), expectedUrls.sort());
+
+	const {flatCalls: logs} = tq.explain(logger);
+	t.deepEqual(logs, warnings);
+});
+
+test('named package', testCli, {
+	arguments_: ['chalk'],
+	urls: ['https://www.npmjs.com/package/chalk'],
+});
+
+test('multiple packages', testCli, {
+	arguments_: ['execa', 'ava'],
+	urls: [
+		'https://www.npmjs.com/package/execa',
+		'https://www.npmjs.com/package/ava',
+	],
+});
 
 for (const flag of ['--github', '-g']) {
-	test(`github: ${flag}`, testCli, [flag]);
-	test(`named package - github: ${flag}`, testCli, [flag, 'chalk']);
-	test(`multiple packages - github: ${flag}`, testCli, [flag, 'execa', 'ava']);
+	test(`github: ${flag}`, testCli, {
+		arguments_: [flag],
+		urls: ['https://github.com/sindresorhus/npm-home'],
+	});
+
+	test(`named package - github: ${flag}`, testCli, {
+		arguments_: [flag, 'chalk'],
+		urls: ['https://github.com/chalk/chalk'],
+	});
+
+	test(`multiple packages - github: ${flag}`, testCli, {
+		arguments_: [flag, 'execa', 'ava'],
+		urls: [
+			'https://github.com/sindresorhus/execa',
+			'https://github.com/avajs/ava',
+		],
+	});
 
 	test(`github - does not error on missing package: ${flag}`, async t => {
 		// https://github.com/npm/validate-npm-package-name#naming-rules
 		const {stderr} = await t.throwsAsync(execa('./cli.js', [flag, '~invalid~']));
 		t.is(stderr, '✖ ~invalid~ - package not found!');
 	});
 
-	test(`github - invalid repository warning: ${flag}`, async t => {
+	test(`github - invalid repository warning: ${flag}`, testOpen, async t => {
 		const {stderr} = await execa('./cli.js', [flag, 'babel-preset-minify']); // From #5
 		t.is(stderr, '✖ The `repository` field in package.json should point to a Git repo and not a website. Please open an issue or pull request on `babel-preset-minify`.');
 	});
+
+	test(`github - falls back to homepage and warns on invalid repository URL: ${flag}`, testCli, {
+		arguments_: [flag, 'foo'],
+		packageJson: {
+			name: 'foo',
+			repository: {url: 'foo'},
+			homepage: 'https://example.com',
+		},
+		urls: ['https://example.com'],
+		warnings: [
+			'✖ The `repository` field in package.json is invalid. Please open an issue or pull request on `foo`.',
+			'⚠ Falling back to `homepage` field.',
+		],
+	});
+
+	test(`github - no repository: ${flag}`, testCli, {
+		arguments_: [flag, 'foo'],
+		packageJson: {name: 'foo'},
+		urls: ['https://www.npmjs.com/package/foo'],
+	});
```

Comment:
Also just realized that this message is incorrect because we fallback to opening the NPM page if the resolved `package.json` doesn't contain a `"repository"` field:

https://github.com/sindresorhus/npm-home/blob/c0e7223bdf35e3b9899af568b05c791824e7be64/cli.js#L45-L51

https://github.com/sindresorhus/npm-home/blob/c0e7223bdf35e3b9899af568b05c791824e7be64/cli.js#L68